imagefetch/imagepull.go

```diff
@@ -116,21 +115,40 @@ func pullImageFromRepos(config config.Config, authConfigs map[string][]docker.Au
 			}
 		}
 
-		var err error
+		// default the doman to docker io.
 		if domain == "" {
-			err = pullSingleImageFromRepo(client, opts, docker.AuthConfiguration{})
-		} else if auth_array, ok := authConfigs[domain]; !ok {
-			err = pullSingleImageFromRepo(client, opts, docker.AuthConfiguration{})
-		} else {
-			for i, auth := range auth_array {
-				err = pullSingleImageFromRepo(client, opts, auth)
-				if err == nil {
-					break
-				} else if i < len(auth_array)-1 {
-					glog.V(5).Infof("Docker image pull(s) failed for service %v docker image %v with auth name %v. Error: %v. Try next auth.", name, service.Image, auth.Username, err)
-				}
+			domain = "docker.io"
+		}
+
+		// get all the auths for this domain or repo.
+		auth_array := []docker.AuthConfiguration{}
+		for k, _ := range authConfigs {
+			a_auth := authConfigs[k]
```

Comment:
Why not:
 for k, a_auth := range authConfigs{ ?

Reply:
I worry about the a_auth changes value as we iterates. I forgot what was the trick for this.

Reply:
The problem with iteration occurs when you take the address of an element and try to use it later. In this case you're appending the array element to another array, so I dont think you have that problem. See this example: https://play.golang.org/p/lc8SzA3LVGx that *I think* verifies that there is no problem.

I think you can use authConfig[k] in the append where you use a_auth right now.